pkg/scalers/datadog_scaler.go

```diff
@@ -304,6 +304,13 @@ func (s *datadogScaler) getQueryResult(ctx context.Context) (float64, error) {
 	for i := 0; i < len(series); i++ {
 		points := series[i].GetPointlist()
 		index := len(points) - 1
+		// Find out the last point != nil
+		for i := len(points) - 1; i >= 0; i-- {
+			if len(points[index]) >= 2 && points[i][1] != nil {
+				index = i
+				break
+			}
+		}
```

Comment:
Should we add this as a feature activated by a flag? I mean, maybe returning 0 or error is correct in some cases and we are hiding that, or maybe not, I don't have expertise with Datadog...

Reply:
good point. Is the original behavior a bug or a feature? 😄

Reply:
@zroubalik Sorry for the late response, after digging deeper this issue, I found the null value in the last data point and the second to last point of query ```sum:trace.express.request.hits{service:foo,env:bar}.as_rate()/avg:kubernetes.cpu.requests{service:foo,env:bar}.rollup(10)``` is because the last data point and the second to last point of the ```sum:trace.express.request.hits{service:foo,env:bar}.as_rate()``` was supposed to be there but it was missing, so without the value of dividend in the query ```A / B```, datadog returns the null value. I spent some time discussing with datadog support

Here is the completed thread of the discussion:
<img width="805" alt="image" src="https://user-images.githubusercontent.com/23068316/206325183-66c59a01-0566-4b86-a569-0969e8f335aa.png">

```json
# query-1: original
# FROM = 1670307226, TO = 1670307286
{
  "status": "ok",
  "resp_version": 1,
  "series": [
    {
      "end": 1670307269000,
      "attributes": {},
      "metric": "trace.express.request.hits",
      "interval": 10,
      "tag_set": [],
      "start": 1670307230000,
      "length": 4,
      "query_index": 0,
      "aggr": "sum",
      "scope": "*",
      "pointlist": [
        [
          1670307230000,
          10062.9
        ],
        [
          1670307240000,
          9680.3
        ],
        [
          1670307250000,
          10414.8
        ],
        [
          1670307260000,
          4839.8
        ]
      ],
      "expression": "sum:trace.express.request.hits{*}.as_rate()",
      "unit": [
        {
          "family": "cache",
          "scale_factor": 1,
          "name": "hit",
          "short_name": null,
          "plural": "hits",
          "id": 39
        },
        {
          "family": "time",
          "scale_factor": 1,
          "name": "second",
          "short_name": "s",
          "plural": "seconds",
          "id": 11
        }
      ],
      "display_name": "trace.express.request.hits"
    }
  ],
  "to_date": 1670307286000,
  "query": "sum:trace.express.request.hits{*}.as_rate()",
  "message": "",
  "res_type": "time_series",
  "times": [],
  "from_date": 1670307226000,
  "group_by": [],
  "values": []
}

# query-2: sleep 30
# FROM = 1670307226, TO = 1670307286
{
  "status": "ok",
  "resp_version": 1,
  "series": [
    {
      "end": 1670307289000,
      "attributes": {},
      "metric": "trace.express.request.hits",
      "interval": 10,
      "tag_set": [],
      "start": 1670307230000,
      "length": 6,
      "query_index": 0,
      "aggr": "sum",
      "scope": "*",
      "pointlist": [
        [
          1670307230000,
          10062.9
        ],
        [
          1670307240000,
          9680.3
        ],
        [
          1670307250000,
          10415.2
        ],
        [
          1670307260000,
          10463.8
        ],
        [
          1670307270000,
          10360.2
        ],
        [
          1670307280000,
          17303.333333333336
        ]
      ],
      "expression": "sum:trace.express.request.hits{*}.as_rate()",
      "unit": [
        {
          "family": "cache",
          "scale_factor": 1,
          "name": "hit",
          "short_name": null,
          "plural": "hits",
          "id": 39
        },
        {
          "family": "time",
          "scale_factor": 1,
          "name": "second",
          "short_name": "s",
          "plural": "seconds",
          "id": 11
        }
      ],
      "display_name": "trace.express.request.hits"
    }
  ],
  "to_date": 1670307286000,
  "query": "sum:trace.express.request.hits{*}.as_rate()",
  "message": "",
  "res_type": "time_series",
  "times": [],
  "from_date": 1670307226000,
  "group_by": [],
  "values": []
}

# query-3: TO = TO + 10s
# FROM = 1670307226, TO = 1670307296
{
  "status": "ok",
  "resp_version": 1,
  "series": [
    {
      "end": 1670307299000,
      "attributes": {},
      "metric": "trace.express.request.hits",
      "interval": 10,
      "tag_set": [],
      "start": 1670307230000,
      "length": 7,
      "query_index": 0,
      "aggr": "sum",
      "scope": "*",
      "pointlist": [
        [
          1670307230000,
          10062.9
        ],
        [
          1670307240000,
          9680.3
        ],
        [
          1670307250000,
          10415.2
        ],
        [
          1670307260000,
          10463.8
        ],
        [
          1670307270000,
          10360.2
        ],
        [
          1670307280000,
          10382
        ],
        [
          1670307290000,
          12534.5
        ]
      ],
      "expression": "sum:trace.express.request.hits{*}.as_rate()",
      "unit": [
        {
          "family": "cache",
          "scale_factor": 1,
          "name": "hit",
          "short_name": null,
          "plural": "hits",
          "id": 39
        },
        {
          "family": "time",
          "scale_factor": 1,
          "name": "second",
          "short_name": "s",
          "plural": "seconds",
          "id": 11
        }
      ],
      "display_name": "trace.express.request.hits"
    }
  ],
  "to_date": 1670307296000,
  "query": "sum:trace.express.request.hits{*}.as_rate()",
  "message": "",
  "res_type": "time_series",
  "times": [],
  "from_date": 1670307226000,
  "group_by": [],
  "values": []
}
```
<img width="839" alt="image" src="https://user-images.githubusercontent.com/23068316/206325406-1998ee24-bab0-49c2-a293-f2a03b9acf91.png">
<img width="834" alt="image" src="https://user-images.githubusercontent.com/23068316/206325476-d134a9e9-635b-4f31-b6df-684d59e16f12.png">
ref: https://docs.datadoghq.com/dashboards/functions/rollup/#rollup-interval-enforced-vs-custom
<img width="845" alt="image" src="https://user-images.githubusercontent.com/23068316/206325572-417ee80c-3771-453e-b4a3-e8e1ec258151.png">

The result with as_rate
![image](https://user-images.githubusercontent.com/23068316/206325768-604bf150-3d94-453f-8495-f41e5079cd09.png)

The result without as_rate
![image](https://user-images.githubusercontent.com/23068316/206325846-549cef67-d04d-4089-b284-990a352895d8.png)

<img width="912" alt="image" src="https://user-images.githubusercontent.com/23068316/206325980-a1800da5-4663-43b5-b411-28e3c6abe8ac.png">

To summarize datadog support's answer:
1. The null value is not a feature, it's just because the last and second to the last points of the dividend ```sum:trace.express.request.hits{*}.as_rate()``` has not been available yet. This PR should address this issue since the newly added logic would return the last point with available value.
2. If we want to get the completed points, instead of using ```now``` for query's time window, we should delay the ```to``` a little bit, like ```now - 30s``` to make sure all data points within the specified time window  are already available. We might need another PR to add a new parameter to let users able to adjust the delay time.
3. The value of the last point could be inaccurate due to the implicit ```rollup(20)```. We might need another PR to always return the second to last point instead for the accuracy or add a new parameter to let users choose if they want to return the last point or the second to last point. What do you think ?

Reply:
@JorTurFer @zroubalik I added two parameters `timeWindowOffset` to fix issue 2 and `lastAvailablePointOffset` to fix issue 3. Not sure if adding new parameters is allowed at the last minute since you plan to release a new version today, but the newly added logic for these two parameters is quite simple, hopefully I can catch up with the release. 

PR for the doc: https://github.com/kedacore/keda-docs/pull/1002

Reply:
Thanks for the investigation, I think we can include this parameters in the release. There's one thing, do you think you are able to write e2e test for this feature?

https://github.com/kedacore/keda/blob/main/tests/scalers/datadog/datadog_test.go

Reply:
okay, let's merge this PR now. could you please open an issue to track this? (adding e2e test for this feature)?